Migration guide for eventing from 1.20.0 to 1.21.0 (#10880)
kyma-project/kyma: docs/migration-guides/1.20-1.21.md

# Migrate from 1.20 to 1.21

## Eventing

> In Kyma 1.21.0, the eventing component has been switched completely from Knative to Kyma internal Subscription custom resource which is based on Nats server. In case the subscribers fail to get triggered by events, check the status of the subscriptions:

Reviewer: ```suggestion
In Kyma 1.21.0, the Eventing component has been switched completely from Knative to Kyma internal Subscription Custom Resource which is based on NATS server. 

In case the subscribers fail to get triggered by events, check the status of the subscriptions:
```

```bash
## Get all subscriptions
kubectl get subscriptions.eventing.kyma-project.io -n <your-namespace>
## Check status sub-resource of the subscription resource
kubectl get subscriptions.eventing.kyma-project.io <subscription-name> -n <your-namespace> -oyaml
```

The upgrade process takes care of migrating triggers to subscriptions and even changing the subscriber URL to sink(subscriber) URL if `svc.cluster.local` suffix is missing and yes, it can break `triggers` which are having subscribers outside the namespace of the trigger(now subscription) or outside the Kyma cluster. This is a known limitation of the new Subscription CRD. The sink of the subscription CR should be a Kubernetes cluster-local URL for a service which is in the same namespace as that of the subscription. The `eventType.value` for each `filter` is of the format: `prefix.application-name.event-type.version`. E.g. `sap.kyma.custom.commerce.order.created.v1`. This is equivalent to a trigger with a following filter:
```
filter:
  attributes:
    eventtypeversion: v1
    source: commerce
    type: order.created
```

> A sample trigger with its equivalent subscription is as follows:

Reviewer: ```suggestion
This is an example of a trigger with its equivalent subscription:
```
